lib/internal/errors.js

```diff
@@ -120,8 +112,13 @@ const prepareStackTrace = (globalThis, error, trace) => {
   //     at function (file)
   //     at file
   let errorString;
-  if (kIsNodeError in error) {
-    errorString = `${error.name} [${error.code}]: ${error.message}`;
+  // Do not use primordials here: we intercept user code here.
+  if (typeof error.toString === 'function' &&
+      error.toString !== MainContextObjectToString) {
+    errorString = error.toString();
```

Comment:
```suggestion
    errorString = String(error);
```
otherwise it won't be guaranteed to be a string, i think?

Reply:
It would be wrong to use a check on `toString` then, as the string value could come from `[Symbol.toPrimitive]`

Reply:
I feel like there is no 100% right or wrong for this case. I opted for falling back to the native toString call in case the return type is not a string.